pkg/gatewayserver/observability.go

```diff
@@ -209,26 +218,44 @@ var gsMetrics = &messageMetrics{
 		},
 		[]string{protocol},
 	),
+	downlinkTxAckForwarded: metrics.NewContextualCounterVec(
```

Comment:
1. Fire `downlinkTxSucceeded` and `downlinkTxFailed` events
2. Consider removing the added `downlinkTxAckForwarded` and `downlinkTxAckDropped` events. I'm not against keeping them, because we can then see the balance between received messages and forwarded/dropped messages and see if they add up, but it should be very clear to the user what is what